include/gsl/pointers

```diff
@@ -119,6 +119,11 @@ private:
     T ptr_;
 };
 
+template <class T>
+auto make_not_null(T&&t) {
+    return gsl::not_null<std::remove_cv_t<std::remove_reference_t<T>>>{t};
```

Comment:
If the purpose of `make_not_null` is just to forward to the `not_null` constructor, isn't `std::forward` the right thing to use here instead?

Reply:
Nice addition, thanks.
And I agree with @neilmacintosh, should it be
```
template <class T>
auto make_not_null(T&&t) {
    return gsl::not_null<std::remove_cv_t<std::remove_reference_t<T>>>{std::forward<T>(t)};
}
```